Initial commit of entry form demo

d4nleonard opened the pull request.

danielleonard-blackbaud committed:
Initial commit of entry form demo

danielleonard-blackbaud committed:
- refactor a bit
- emphasize addInEvent.done in blocking events

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

michael-tims commented on src/app/shared/services/entry-form.service.ts:
```diff
@@ -0,0 +1,46 @@
+  public logCancel(): Observable<boolean> {
+    return of(true).pipe(
+      concatMap(item => of(item).pipe(
+        delay(3000)
```
Maybe we can shorten these delays down a bit, maybe to 1 second?  It looks like for the cancel we may need to show the wait indicator from the gift entry form code.  I can work on doing that separately.

d4nleonard replied:
Making this change now

d4nleonard replied:
Made change down to 1 second of delay

michael-tims commented on src/app/my-tab/my-tab.component.ts:
```diff
@@ -0,0 +1,140 @@
+    this.captureGiftFormCancelEvent();
+
+    // capture the save event triggered from the gift form through the add in client
+    this.captureGiftFormSaveEvent();
```
Maybe rename these to "handle...Event" rather than "capture...Event".  Handle is the naming we use in the API.

d4nleonard replied:
Making this change now

d4nleonard replied:
Renamed methods and comments to use 'handle' and not 'capture'

michael-tims commented on src/app/my-tab/my-tab.component.ts:
```diff
@@ -0,0 +1,140 @@
+      // initialize the tab
+      args.ready({
+        showUI: true,
+        title: `My entry form Add-in tab`
```
May want to shorten this name of the tab. Even just taking out "Add-in" so it becomes "Entry form demo" or something like that. Don't worry about this if you've already started creating screenshots for the doc.

d4nleonard replied:
Making this change now

d4nleonard replied:
Renamed add-in tab to "Entry form demo"

michael-tims reviewed (commented):
Looking really good!  Just a couple suggestions.

danielleonard-blackbaud committed:
- Moving things around to make the docs easier
- Updating needed packages
- Updating module files

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

danielleonard-blackbaud committed:
- rename tab
- rename capture to handle for event methods
- change the delay to 1 seconds for mock service

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

danielleonard-blackbaud committed:
update comments to say handle instead of capture

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

michael-tims reviewed (approved):
Looks good!

danielleonard-blackbaud committed:
Update readme file to be more specific

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

danielleonard-blackbaud committed:
Change for proper branding

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

lindsey-rix commented on README.md:
```diff
@@ -1 +1,14 @@
\ No newline at end of file
+# skyux-spa-addin-entry-form-demo
+
+This SPA demonstrates a SKY Add-in implementation using SKY UX. The example uses the Razor's Edge NXT Gift Entry form and side loads a vertical add-in tab in the same modal dialog tabset.
```
Only thing I noticed is that Raiser's Edge is misspelled.

d4nleonard replied:
Fixed the spelling error. Thanks!

danielleonard-blackbaud committed:
adjust comments

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)

danielleonard-blackbaud committed:
Fix spelling error

blackbaud-ado commented:
Preview build at [https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit](https://host.nxt.blackbaud.com/addin-entry-form-demo?_sp.spa=pr-initial-commit)
